src/base/ini_parser.c

```diff
@@ -559,6 +559,16 @@ ini_parse_line (char *input_line, char *section, char *key, char *value)
       /* Generate syntax error */
       status = LINE_ERROR;
     }
+
+  if (strcmp (key, "shard_db_password") == 0 && strlen (value) > 0)
```

Comment:
It seems like a not good way to compare a specific key value (shard_db_password) in the function. 
Please consider modify it by the way ignoring the # inside single quotes.